specs/agents/tracing-instrumentation-aws.md

```diff
@@ -0,0 +1,57 @@
+## AWS services spans
+
+We describe how to instrument some of AWS' services in this document.
+Some of the services can use existing specs. When there are differences or additions, they have been noted below.
+
+### S3 (Simple Storage Service)
+
+AWS Simple Storage Service offers object storage via a REST API. The objects are organized into buckets, which are 
+themselves organized into regions.
+
+- `span.name`: The span name should follow this pattern: `S3 <OperationName> <bucket-name>`. For example,
+`S3 GetObject my-bucket`. Note that the operation name is in CamelCase.
+- `span.type`: `storage`
+- `span.subtype`: `s3`
+- `span.action`: The operation name in CamelCase. For example ‘GetObject’.
+
+#### Span context fields
+
+- **`context.destination.address`**: optional. Not available in some cases. Only set if the actual connection is available.
+- **`context.destination.port`**: optional. Not available in some cases. Only set if the actual connection is available.
+- **`context.destination.region`**: mandatory. The AWS region where the bucket is.
+- **`context.destination.service.name`**: mandatory. Use `s3`
+- **`context.destination.service.resource`**: optional. The bucket name, if available. The s3 API allows either the
+bucket name or an Access Point to be provided when referring to a bucket. Access Points can use either slashes or colons.
+When an Access Point is provided, the access point name preceded by `accesspoint/` or `accesspoint:` should be extracted.
+For example, given an Access Point such as `arn:aws:s3:us-west-2:123456789012:accesspoint/myendpointslashes`, the agent
+extracts `accesspoint/myendpointslashes`. Given an Access Point such as
+`arn:aws:s3:us-west-2:123456789012:accesspoint:myendpointcolons`, the agent extracts `accesspoint:myendpointcolons`.
+- **`context.destination.service.type`**: mandatory. Use `storage`.
+
+### DynamoDB
+
+AWS DynamoDB is a document database so instrumenting it will follow the [db spec](tracing-instrumentation-db.md). 
```

Comment:
Would it make sense to also spec out the span name, as the current db spec doesn't have any guidance on that? Maybe something like

    DynamoDB <ActionName> <TableName>

which would look like

    DynamoDB PutItem Books

Also, should we capture the "query" (if you want to call it that), and if yes, only for some calls? e.g. we probably don't want to capture `PutItem`, but maybe `GetItem` or `DeleteItem` (I don't have enough experience with DynamoDB to say if the query has any influence on the performance).

Reply:
@beniwohli I've updated the spec. Let me know if it makes sense, thanks!

Reply:
Looks good!